cmd/workspace/apps/run-local.go

```diff
@@ -0,0 +1,205 @@
+package apps
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/databricks/cli/cmd/root"
+	"github.com/databricks/cli/libs/appproxy"
+	"github.com/databricks/cli/libs/apps"
+	"github.com/databricks/cli/libs/auth"
+	"github.com/databricks/cli/libs/cmdctx"
+	"github.com/databricks/cli/libs/cmdio"
+	"github.com/spf13/cobra"
+)
+
+var (
+	port               int
+	debug              bool
+	prepareEnvironment bool
+	entryPoint         string
+	customEnv          []string
+)
+
+func newRunLocal() *cobra.Command {
+	cmd := &cobra.Command{}
+
+	cmd.Use = "run-local"
+	cmd.Short = `Run an app locally`
+	cmd.Long = `Run an app locally.
+
+	  This command starts an app locally.`
+
+	cmd.Flags().IntVar(&port, "port", 8001, "Port on which to run the app")
+	cmd.Flags().BoolVar(&debug, "debug", false, "Enable debug mode")
+	cmd.Flags().BoolVar(&prepareEnvironment, "prepare-environment", false, "Prepares the environment for running the app. Requires 'uv' to be installed.")
+	cmd.Flags().StringSliceVar(&customEnv, "env", nil, "Set environment variables")
+	cmd.Flags().StringVar(&entryPoint, "entry-point", "", "Specify the custom entry point with configuration (.yml file) for the app. Defaults to app.yml")
+
+	cmd.PreRunE = root.MustWorkspaceClient
+	cmd.RunE = func(cmd *cobra.Command, args []string) (err error) {
+		ctx := cmd.Context()
+		w := cmdctx.WorkspaceClient(ctx)
+		workspaceId, err := w.CurrentWorkspaceID(ctx)
+		if err != nil {
+			return err
+		}
+
+		cwd, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+
+		config := apps.NewConfig(w.Config.Host, workspaceId, cwd)
+		if entryPoint != "" {
+			config.AppSpecFiles = []string{entryPoint}
+		}
+		spec, err := apps.ReadAppSpecFile(config)
+		if err != nil {
+			return err
+		}
+
+		cfg := cmdctx.ConfigUsed(cmd.Context())
+		env := auth.ProcessEnv(cfg)
+		if cfg.Profile != "" {
+			env = append(env, "DATABRICKS_CONFIG_PROFILE="+cfg.Profile)
+		}
+
+		for _, envVar := range spec.EnvVars {
+			if envVar.Value != nil {
+				env = append(env, envVar.Name+"="+*envVar.Value)
+			}
+
+			if envVar.ValueFrom != nil {
+				if os.Getenv(envVar.Name) != "" {
```

Comment:
Can use `env.Lookup(ctx, envVar.Name)`.

This looks similar to the code in `libs/apps/env.go`. Any opportunity to reuse/share?